deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/base/MatchIndex.java

```diff
@@ -287,46 +289,68 @@ private CompareResult compareConfigurablePathSegment(@NonNull byte[] inboundValu
 
   @Nullable
   private CompareResult compareComponentParam(
-    int valueStartPos, int valueLength, @NonNull byte[] inboundValue
+    int valueStartPos, int valueLength, @NonNull byte[] valueToMatch, char[] delimiter
   ) {
+    int valueToMatchLength = valueToMatch.length;
     if ((
       //Per com.airbnb.deeplinkdispatch.DeepLinkEntryTest.testEmptyParametersNameDontMatch
       //We should not return a match if the param (aka placeholder) is empty.
-      byteArray[valueStartPos] == '{' && byteArray[valueStartPos + 1] == '}'
+      byteArray[valueStartPos] == delimiter[0] && byteArray[valueStartPos + 1] == delimiter[1]
     ) || (
       //Per com.airbnb.deeplinkdispatch.DeepLinkEntryTest.testEmptyPathPresentParams
       //We expect an empty path to not be a match
-      inboundValue.length == 0
+      valueToMatchLength == 0
     )) {
       return null;
     }
-    // i index over inboundValue array forward
-    // j index over search byte arrays inboundValue element forward
-    // k index over inboundValue array backward
-    for (int i = 0; i < inboundValue.length; i++) {
-      if (byteArray[valueStartPos + i] == '{') {
+    // i index over valueToMatch array forward
+    // j index over search byte arrays valueToMatch element forward
+    // k index over valueToMatch array backward
+    for (int i = 0; i < valueToMatchLength; i++) {
+      // Edge case. All chars before placeholder match. And there are no more chars left in
+      // input string. But there are still chars left in value. In that case we need to look
+      // at the rest of the value as there might chars after the placeholder (e.g. pre{ph}post)
+      //or a placeholder that matches the empty string. e.g.
+      boolean fullyMatchedButThereAreMoreCharsInValue =
+        byteArray[valueStartPos + i] == valueToMatch[i]
+          && i == valueToMatchLength - 1
+          && valueLength > valueToMatchLength;
```

Comment:
given how much you've been working in this class it would be great to convert it to kotlin

Reply:
I did that originally and had performance problems making it about 60% slower. So I moved it back to java. I can eventually look at that again, but for now I think it is just not worth it, also given how low level that is.

I know back then we did think these were type conversions happening but I did not want to hunt them down yet. :-)